custom/enikshay/reports/filters.py

```diff
@@ -47,6 +47,8 @@ def selected(self):
     @classmethod
     def get_value(cls, request, domain):
         selected = super(EnikshayLocationFilter, cls).get_value(request, domain)
+        if not request.couch_user.has_permission(domain, 'access_all_locations'):
+            selected += request.couch_user.get_location_ids(domain)
```

Comment:
Won't this sometimes include more locations than are selected?  If I'm assigned to Suffolk and I select "Boston", I'd expect to see only results in Boston, but I think with this I'd see results in both Boston AND Suffolk.  I think the user's location ids should only be added if they haven't selected anything, and if they have selected something, we should make sure the selections are accessible (in case they modified the request params or the choice provider was broken).